infra/templates/az-svc-data-integration-mlw/app.tf

```diff
@@ -0,0 +1,126 @@
+resource "azurerm_resource_group" "app_rg" {
+  name     = local.app_rg_name
+  location = local.region
+}
+
+# Note: this should be uncommented for production scenarios. It is commented
+#       to support a teardown after deployment for the CICD pipeline.
+# resource "azurerm_management_lock" "app_rg_lock" {
+#   name       = local.app_rg_lock
+#   scope      = azurerm_resource_group.app_rg.id
+#   lock_level = "CanNotDelete"
+
+#   lifecycle {
+#     prevent_destroy = true
+#   }
+# }
+
+module "network" {
+  source              = "../../modules/providers/azure/network"
+  vnet_name           = local.vnet_name
+  resource_group_name = azurerm_resource_group.app_rg.name
+  address_space       = var.address_space
+  subnets             = local.subnets
+}
+
+module "container_registry" {
+  source                           = "../../modules/providers/azure/container-registry"
+  container_registry_name          = local.acr_name
+  resource_group_name              = azurerm_resource_group.app_rg.name
+  container_registry_admin_enabled = false
+  // Note: only premium ACRs allow configuration of network access restrictions
+  container_registry_sku = var.container_registry_sku
+  subnet_id_whitelist    = module.network.subnet_ids
+}
+
+module "app_insights" {
+  source                           = "../../modules/providers/azure/app-insights"
+  service_plan_resource_group_name = azurerm_resource_group.app_rg.name
+  appinsights_name                 = local.ai_name
+  appinsights_application_type     = "web"
+}
+
+
+module "func_app_service_plan" {
+  source              = "../../modules/providers/azure/service-plan"
+  resource_group_name = azurerm_resource_group.app_rg.name
+  service_plan_name   = local.func_app_sp_name
+  # scaling_rules       = var.scaling_rules
```

Comment:
If this is not needed it can be removed, along with any referenced variables that are not necessary.